zkevm-circuits/src/evm_circuit/execution/error_oog_sload_sstore.rs

```diff
@@ -0,0 +1,585 @@
+                block.rws[step.rw_indices[6]].storage_value_aux();
+            let gas_cost =
+                cal_sstore_gas_cost_for_assignment(value, value_prev, original_value, is_warm);
+            (value, value_prev, original_value, gas_cost)
+        } else {
+            let gas_cost = cal_sload_gas_cost_for_assignment(is_warm);
+            (U256::zero(), U256::zero(), U256::zero(), gas_cost)
+        };
+
+        log::debug!(
+            "ErrorOutOfGasSloadSstore: is_sstore = {}, gas_left = {}, gas_cost = {}, gas_sentry = {}",
+            is_sstore,
+            step.gas_left,
+            gas_cost,
+            if is_sstore { GasCost::SSTORE_SENTRY.0 } else { 0 },
+        );
+
+        self.opcode
+            .assign(region, offset, Value::known(F::from(opcode.as_u64())))?;
+        self.tx_id
+            .assign(region, offset, Value::known(F::from(tx.id as u64)))?;
+        self.is_static
+            .assign(region, offset, Value::known(F::from(call.is_static as u64)))?;
+        self.callee_address.assign(
+            region,
+            offset,
+            Value::known(
+                call.callee_address
+                    .to_scalar()
+                    .expect("unexpected Address -> Scalar conversion failure"),
+            ),
+        )?;
+        self.phase2_key
+            .assign(region, offset, region.word_rlc(key))?;
+        self.phase2_value
+            .assign(region, offset, region.word_rlc(value))?;
+        self.phase2_value_prev
+            .assign(region, offset, region.word_rlc(value_prev))?;
+        self.phase2_original_value
+            .assign(region, offset, region.word_rlc(original_value))?;
+        self.is_warm
+            .assign(region, offset, Value::known(F::from(is_warm as u64)))?;
+        self.rw_counter_end_of_reversion.assign(
+            region,
+            offset,
+            Value::known(F::from(call.rw_counter_end_of_reversion as u64)),
+        )?;
+        self.is_sstore.assign(
+            region,
+            offset,
+            F::from(opcode.as_u64()) - F::from(OpcodeId::SSTORE.as_u64()),
+        )?;
+        self.sstore_gas_cost
+            .assign(region, offset, value, value_prev, original_value, is_warm)?;
+        self.insufficient_gas_cost.assign_value(
+            region,
+            offset,
+            Value::known(F::from(step.gas_left)),
+            Value::known(F::from(gas_cost)),
+        )?;
+        self.insufficient_gas_sentry.assign_value(
+            region,
+            offset,
+            Value::known(F::from(step.gas_left)),
+            Value::known(F::from(GasCost::SSTORE_SENTRY.0.checked_add(1).unwrap())),
+        )?;
+        self.restore_context.assign(
+            region,
+            offset,
+            block,
+            call,
+            step,
+            // Additional one stack pop and one account storage read for SSTORE.
+            7 + if is_sstore { 2 } else { 0 },
```

Comment:
Tried to fix as `7 + is_sstore * 2` but failed to compile. It seems that I set `is_sstore` as a boolean in `assign_exec_step`:
```
let is_sstore = opcode == OpcodeId::SSTORE;
```
And I use this boolean as `if is_sstore` in multiple places (otherwise it should be replaced with `if is_sstore == 1`).

Reply:
Fixed to `7 + usize::from(is_sstore) * 2` in [9fc3451](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1147/commits/9fc345150ca59c6a8e04e3d784e506b161d479ee).